src/bsoncxx/builder/list/document.hpp

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
```

Comment:
Same comment as with the array builder about moving function definitions to a `.cpp` file instead, and tagging remaining function definitions in the header with BSONCXX_INLINE.

Reply:
I'm still working on this but all other comments should be addressed.